src/ipc/ipc4/handler.c

```diff
@@ -396,15 +396,16 @@ static void ipc_compound_msg_done(uint32_t msg_id, int error)
 	}
 }
 
+/* wait for IPCs to complete on other cores and be nice to any LL work */
 static int ipc_wait_for_compound_msg(void)
 {
-	int try_count = 30;
+	int try_count = 30; /* timeout out is 30 x 10ms so 300ms for IPC */
 
 	while (atomic_read(&msg_data.delayed_reply)) {
-		k_sleep(Z_TIMEOUT_US(250));
+		k_sleep(Z_TIMEOUT_MS(10));
```

Comment:
Delayed IPCs are expected to take time for the DSP to process and there is no critical urgency on the host side to get the response. i.e. it does not matter if DSP replies to host on nearest 1ms or nearest 10ms for long IPC messages.

However, on the DSP side if we schedule a wake (timer IRQ) every 250uS or every 1ms then we can spend more time scheduling and context switching instead of audio processing. This change means we do 40x less effort around sleep context switches / timer IRQs.